app/controllers/users/omniauth_callbacks_controller.rb

```diff
@@ -27,17 +28,29 @@ def idme
       return redirect_to params["after_login"]
     end
 
-    should_delete_duplicate_intake = session[:intake_id] && (is_returning_user || is_new_spouse)
+    # A new intake is created before sign in so that we can save answers to questions that are asked before sign in.
+    # If a returning user signs in, we want to delete this new intake because the user will already have an intake.
+    # TODO: We also want to do this for a new spouse - why??
+    # We want to skip this for a spouse who came to the sign in page with an authenticate-later-link because the intake
+    # in the session in that case is the one corresponding to the token in the link, which we will want to keep so that we
+    # can associate it with the new spouse user.
+    should_delete_duplicate_intake = session[:intake_id] && (is_returning_user || is_new_spouse) && !used_spouse_auth_only_link
     if should_delete_duplicate_intake
       intake_from_session = Intake.find(session[:intake_id])
       intake_from_session.destroy! if intake_from_session.users.count == 0
       session[:intake_id] = nil
     end
 
     if is_new_spouse
+      if used_spouse_auth_only_link
+        # TODO: could alternatively use Intake.find_by_id(session[:intake_id]) here just to be sure
```

Comment:
opinions? what do we think are the odds that someone is already signed in on the device where this person is visiting the authorize spouse link? and that the person signed in is not the primary? that's the only case in which this would go wrong